src/common/pmix_attributes.c

```diff
@@ -58,13 +59,49 @@ static void atrkdes(pmix_attribute_trk_t *p)
 }
 static PMIX_CLASS_INSTANCE(pmix_attribute_trk_t, pmix_list_item_t, atrkcon, atrkdes);
 
+/* define a couple of internal attributes */
+static pmix_regattr_input_t internals[] = {
+    {.index = PMIX_INDEX_BOUNDARY, .name = "PMIX_BFROPS_MODULE", .string = "pmix.bfrops.mod", .type = PMIX_STRING,
+        .description = (char *[]){"name of bfrops plugin in-use by a given nspace"}},
+
+    {.index = PMIX_INDEX_BOUNDARY+1, .name = "PMIX_PNET_SETUP_APP", .string = "pmix.pnet.setapp", .type = PMIX_BYTE_OBJECT,
+            .description = (char *[]){"blob containing info to be given to pnet framework on remote nodes", NULL}},
+
+    {.index = UINT32_MAX, .name = "", .string = "", .type = PMIX_POINTER, .description = (char *[]){"NONE"}}
+};
+
 PMIX_EXPORT void pmix_init_registered_attrs(void)
 {
+    size_t n;
+    pmix_regattr_input_t *p;
+
     if (!initialized) {
         PMIX_CONSTRUCT(&client_attrs, pmix_list_t);
         PMIX_CONSTRUCT(&server_attrs, pmix_list_t);
         PMIX_CONSTRUCT(&host_attrs, pmix_list_t);
         PMIX_CONSTRUCT(&tool_attrs, pmix_list_t);
+
+        /* cycle across the dictionary and load a hash
+         * table with translations of key -> index */
+        for (n=0; UINT32_MAX != dictionary[n].index; n++) {
+            p = (pmix_regattr_input_t*)pmix_malloc(sizeof(pmix_regattr_input_t));
+            p->index = dictionary[n].index;
+            p->name = strdup(dictionary[n].name);
+            p->string = strdup(dictionary[n].string);
+            p->type = dictionary[n].type;
+            p->description = pmix_argv_copy(dictionary[n].description);
+            pmix_hash_register_key(p->index, p);
+        }
+        /* include the internals */
+        for (n=0; UINT32_MAX != internals[n].index; n++) {
+            p = (pmix_regattr_input_t*)pmix_malloc(sizeof(pmix_regattr_input_t));
+            p->index = internals[n].index;
+            p->name = strdup(internals[n].name);
+            p->string = strdup(internals[n].string);
```

Comment:
I've only been able to reproduce the crash with `--disable-debug` builds (if I `--enable-debug` then it does not crash). But the stack trace points to here:
```
(gdb) bt
#0  0x00007fffb5e2e6f4 in __strlen_power8 () from /lib64/libc.so.6
#1  0x00007fffb5e158e4 in strdup () from /lib64/libc.so.6
#2  0x00007fffb66304c0 in pmix_init_registered_attrs () from /workspace/exports/pmix/lib/libpmix.so.0
#3  0x00007fffb65d3844 in pmix_rte_init () from /workspace/exports/pmix/lib/libpmix.so.0
#4  0x00007fffb659fa1c in PMIx_server_init () from /workspace/exports/pmix/lib/libpmix.so.0
#5  0x00007fffb68abbb8 in pmix_server_init () at prted/pmix/pmix_server.c:810
#6  0x00007fffb6944bc8 in rte_init (argc=2, argv=0x10000ff0350) at ess_hnp_module.c:214
#7  0x00007fffb68822a0 in prte_init (pargc=0x7fffd1038db0, pargv=0x7fffd1038da8, flags=4 '\004') at runtime/prte_init.c:315
#8  0x0000000010007354 in main (argc=2, argv=0x7fffd103a4c8) at prte.c:515
```

Reply:
I see. The problem is with this line in the first loop:
```
            p->description = pmix_argv_copy(dictionary[n].description);
```
and this part of the generated dictionary:
```
static pmix_regattr_input_t dictionary[] = {
    {.index = 0, .name = "PMIX_ATTR_UNDEF", .string = "pmix.undef", .type = PMIX_POINTER,
     .description = (char *[]){"NONE"}},
```
Notice that the description is not `NULL` terminated.